### DISCUSSION TITLE
LSP autocompletion for heading IDs?

### DISCUSSION THREAD

**Discussion by dzintars** [zk-org/zk-nvim#53] [Help]
Hi all. REALLY impressive tooling! :)
I have a question about Heading ID autocompletion?
Is it possible?
I know that if many nested levels i a single note breaks the "atomicity rule", but still..
![image](https://user-images.githubusercontent.com/547420/171753210-46f63478-e844-48fd-af32-58ccd7627d13.png)
There is an image of what i mean. That Tmux note contains few sections, i would like to link to.
This is how it looks in Obsidian:
![Frame 1](https://user-images.githubusercontent.com/547420/171755261-f1a146a3-7f66-48c5-beb9-6fa8b1a70ccd.png)
I believe `block` references are Obsidian specific only so i don't care much about them.
But Heading IDs would be helpful to have after typing `#`. Currently it returns list of tags.

**Comment by mickael-menu** [comment]
I opened an issue from your discussion in `zk`, as this feature would be entirely in the LSP server and `zk` codebase: https://github.com/mickael-menu/zk/issues/224
